src/java.base/share/classes/jdk/internal/classfile/impl/TemporaryConstantPool.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.impl;
+
+import jdk.internal.classfile.*;
+import jdk.internal.classfile.constantpool.ClassEntry;
+import jdk.internal.classfile.constantpool.ConstantDynamicEntry;
+import jdk.internal.classfile.constantpool.ConstantPool;
+import jdk.internal.classfile.constantpool.ConstantPoolBuilder;
+import jdk.internal.classfile.constantpool.DoubleEntry;
+import jdk.internal.classfile.constantpool.FieldRefEntry;
+import jdk.internal.classfile.constantpool.FloatEntry;
+import jdk.internal.classfile.constantpool.IntegerEntry;
+import jdk.internal.classfile.constantpool.InterfaceMethodRefEntry;
+import jdk.internal.classfile.constantpool.InvokeDynamicEntry;
+import jdk.internal.classfile.constantpool.LoadableConstantEntry;
+import jdk.internal.classfile.constantpool.LongEntry;
+import jdk.internal.classfile.constantpool.MemberRefEntry;
+import jdk.internal.classfile.constantpool.MethodHandleEntry;
+import jdk.internal.classfile.constantpool.MethodRefEntry;
+import jdk.internal.classfile.constantpool.MethodTypeEntry;
+import jdk.internal.classfile.constantpool.ModuleEntry;
+import jdk.internal.classfile.constantpool.NameAndTypeEntry;
+import jdk.internal.classfile.constantpool.PackageEntry;
+import jdk.internal.classfile.constantpool.PoolEntry;
+import jdk.internal.classfile.constantpool.StringEntry;
+import jdk.internal.classfile.constantpool.Utf8Entry;
+
+import java.lang.constant.MethodTypeDesc;
+import java.util.Collections;
+import java.util.List;
+
+import static jdk.internal.classfile.Classfile.TAG_METHODTYPE;
+
+public final class TemporaryConstantPool implements ConstantPoolBuilder {
+    private static final Options options = new Options(Collections.emptyList());
+
+    private TemporaryConstantPool() {};
```

Comment:
If I understand correctly, this constant pool is just "fake" in the sense that it's a builder which creates constant pool entries w/o storing them anywhere. This seems to be used in places where e.g. we need to convert a ClassDesc to a ClassEntry, probably so that the implementation of e.g. attributes can be defined in terms of constant pool entries, even when "unbounded". E.g. this is a trick which avoids having completely different representations for bound and unbound elements - correct?

Reply:
Yes, exactly :)